commands/scp.go

```diff
@@ -148,6 +150,11 @@ func getInfoForScpArg(hostAndPath string, hostInfoLoader HostInfoLoader) (HostIn
 	}
 
 	args := []string{}
+	port, err := hostInfo.GetSSHPort()
+	if err == nil && port > 0 {
+		args = append(args, "-P", fmt.Sprintf("%v", port))
+	}
```

Comment:
Here is what I test the function.

From local to remote

```bash
$ bin/docker-machine -D scp ./README.md d2:./
Docker Machine Version:  0.7.0, build 2943cdd
Found binary path at bin/docker-machine
Launching plugin server for driver digitalocean
Plugin server listening at address 127.0.0.1:51671
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(d2) Calling .GetSSHPort
(d2) Calling .GetSSHKeyPath
(d2) Calling .GetSSHKeyPath
(d2) Calling .GetIP
(d2) Calling .GetSSHUsername
{/usr/bin/scp [/usr/bin/scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -3 -P 232 ./README.md root@138.68.50.80:./] []  <nil> <nil> <nil> [] <nil> <nil> <nil> <nil> <nil> false [] [] [] [] <nil> <nil>}
README.md                                                                                                            100% 6375     6.2KB/s   00:00
Making call to close driver server
(d2) Calling .Close
Successfully made call to close driver server
Making call to close connection to plugin binary
(d2) DBG | Closing plugin on server side
```

From remote to local

```bash
$ bin/docker-machine -D scp d2:./README.md ./README.md.2
Docker Machine Version:  0.7.0, build 2943cdd
Found binary path at bin/docker-machine
Launching plugin server for driver digitalocean
Plugin server listening at address 127.0.0.1:51726
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(d2) Calling .GetSSHPort
(d2) Calling .GetSSHKeyPath
(d2) Calling .GetSSHKeyPath
(d2) Calling .GetIP
(d2) Calling .GetSSHUsername
{/usr/bin/scp [/usr/bin/scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -3 -P 232 root@138.68.50.80:./README.md ./README.md.2] []  <nil> <nil> <nil> [] <nil> <nil> <nil> <nil> <nil> false [] [] [] [] <nil> <nil>}
README.md                                                                                                            100% 6375     6.2KB/s   00:01
Making call to close driver server
(d2) Calling .Close
Successfully made call to close driver server
Making call to close connection to plugin binary
```